src/pgapp_worker.erl

```diff
@@ -115,6 +96,17 @@ middle_man_transaction(Pool, Fun, Timeout) ->
             {error, Reason}
     end.
 
+wrap_timeout(Fun) ->
+    fun(Worker) ->
+            try
+                Fun(Worker)
+            catch
+                Type:Error ->
```

Comment:
Mostly yes, given that the current code can return incapacitated workers to the pool which can cause serious issues I think killing a worker too many might be better then killing a worker too few.

But all the calls wrapped are nothing more then a `gen_server:call` so the only other exception possibly being thrown I can think of are all exits anyway.